corehq/apps/users/tests/test_role_utils.py

```diff
@@ -118,3 +126,69 @@ def _delete_presets(self):
         for role in UserRole.objects.get_by_domain(self.domain):
             if role.id != self.role1.id:
                 role.delete()
+
+
+class TestCommcareAnalyticsRolesByUser(TestCase):
+
+    DOMAIN = "domain1"
+    USERNAME = "username"
+    PASSWORD = "***"
+
+    @classmethod
+    def setUpClass(cls):
+        cls.domain = create_domain(cls.DOMAIN)
+        cls.user = WebUser.create("domain1", cls.USERNAME, cls.PASSWORD, None, None)
+
+        cls.hq_no_cca_role = cls.create_role("No CCA role", analytics_roles=None)
+
+        cls.limited_cca_roles = [
+            COMMCARE_ANALYTICS_GAMMA,
+            COMMCARE_ANALYTICS_SQL_LAB,
+        ]
+        cls.hq_limited_cca_role = cls.create_role("Limited CCA role", analytics_roles=cls.limited_cca_roles)
+
+    @classmethod
+    def tearDownClass(cls):
+        for role in UserRole.objects.get_by_domain(cls.domain):
+            role.delete()
+
+        cls.user.delete(deleted_by_domain=cls.domain.name, deleted_by=None)
+        cls.domain.delete()
+
+    def test_user_domain_does_not_have_flag_enabled(self):
+        analytics_roles = get_commcare_analytics_roles_for_user_domain(self.user, self.DOMAIN)
+        self.assertTrue("domain1" not in analytics_roles)
+
+    @flag_enabled('SUPERSET_ANALYTICS')
+    def test_admin_user(self):
+        self.user.domain_memberships[0].is_admin = True
+        self.assertTrue(self.user.get_domain_membership(self.DOMAIN).is_admin)
```

Comment:
Wanted to understand why is this test needed ?
nit: SRP - Would suggest to skip this test if not really needed as we are testing two different things here. Same for other test cases.

Reply:
SRP is?

Reply:
Single Responsibility Principle

Reply:
Google said "Suggested Retail Price" :D

Reply:
Ah, technically not necessary. More just a way of saying "see, it works", but it's really redundant. Will update!